cmd/bastion/aws/create.go

```diff
@@ -0,0 +1,549 @@
+package aws
+
+import (
+	"context"
+	"fmt"
+	"io/ioutil"
+	"os"
+	"os/signal"
+	"regexp"
+	"syscall"
+	"time"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/aws/awserr"
+	"github.com/aws/aws-sdk-go/service/ec2"
+	"github.com/spf13/cobra"
+
+	"k8s.io/apimachinery/pkg/types"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/util/retry"
+
+	hyperv1 "github.com/openshift/hypershift/api/v1alpha1"
+	awsutil "github.com/openshift/hypershift/cmd/infra/aws/util"
+	"github.com/openshift/hypershift/cmd/util"
+)
+
+type createBastionOpts struct {
+	Namespace          string
+	Name               string
+	InfraID            string
+	Region             string
+	SSHKeyFile         string
+	AWSCredentialsFile string
+	Wait               bool
+}
+
+func NewCreateCommand() *cobra.Command {
+	opts := &createBastionOpts{
+		Namespace: "clusters",
+		Wait:      true,
+	}
+
+	cmd := &cobra.Command{
+		Use:          "aws",
+		Short:        "Creates AWS bastion instance",
+		SilenceUsage: true,
+	}
+	cmd.Flags().StringVar(&opts.Namespace, "namespace", opts.Namespace, "The namespace of the hostedcluster")
+	cmd.Flags().StringVar(&opts.Name, "name", opts.Name, "The name of the hostedcluster")
+	cmd.Flags().StringVar(&opts.InfraID, "infra-id", opts.InfraID, "The infra ID to use for creating the bastion")
+	cmd.Flags().StringVar(&opts.Region, "region", opts.Region, "The region to use for creating the bastion")
+	cmd.Flags().StringVar(&opts.SSHKeyFile, "ssh-key-file", opts.SSHKeyFile, "File with public SSH key to use for bastion instance")
+	cmd.Flags().StringVar(&opts.AWSCredentialsFile, "aws-creds", opts.AWSCredentialsFile, "File with AWS credentials")
+	cmd.Flags().BoolVar(&opts.Wait, "wait", opts.Wait, "Wait for instance to be running")
+
+	cmd.MarkFlagRequired("ssh-key-file")
```

Comment:
Can we maybe default this to the value of the referenced secret on the hosted cluster if set, similiar to region and infraID?